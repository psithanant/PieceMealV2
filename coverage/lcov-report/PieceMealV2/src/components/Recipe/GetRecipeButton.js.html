<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for PieceMealV2/src/components/Recipe/GetRecipeButton.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../index.html">All files</a> / <a href="index.html">PieceMealV2/src/components/Recipe</a> GetRecipeButton.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>2/2</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Button, Text } from 'native-base';
import Icon from 'react-native-vector-icons/MaterialCommunityIcons';
import { getRecipes } from '../../actions/index';
&nbsp;
export class GetRecipeButton extends Component {
  render() {
    return (
      &lt;Button
        onPress={this.props.getAllRecipes}
        style={styles.buttonStyle}
        iconLeft
      &gt;
        &lt;Icon name='food-variant' size={35} /&gt;
        &lt;Text&gt;See All Recipes&lt;/Text&gt;
      &lt;/Button&gt;
    );
  }
}
&nbsp;
const styles = {
  buttonStyle: {
    marginLeft: 5,
    marginRight: 5
  }
};
&nbsp;
export default connect(null, { getRecipes })(GetRecipeButton);
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="http://istanbul-js.org/" target="_blank">istanbul</a> at Mon Jun 19 2017 17:51:46 GMT-0700 (PDT)
</div>
</div>
<script src="../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../sorter.js"></script>
</body>
</html>
