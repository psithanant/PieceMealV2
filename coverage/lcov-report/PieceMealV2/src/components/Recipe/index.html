<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for PieceMealV2/src/components/Recipe</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../index.html">All files</a> PieceMealV2/src/components/Recipe
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">32.2% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>38/118</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">33.33% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>8/24</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">15.94% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>11/69</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">32.2% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>38/118</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file low" data-value="Create.js"><a href="Create.js.html">Create.js</a></td>
	<td data-value="11.43" class="pic low"><div class="chart"><div class="cover-fill" style="width: 11%;"></div><div class="cover-empty" style="width:89%;"></div></div></td>
	<td data-value="11.43" class="pct low">11.43%</td>
	<td data-value="35" class="abs low">4/35</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="27" class="abs low">0/27</td>
	<td data-value="11.43" class="pct low">11.43%</td>
	<td data-value="35" class="abs low">4/35</td>
	</tr>

<tr>
	<td class="file medium" data-value="Fab.js"><a href="Fab.js.html">Fab.js</a></td>
	<td data-value="75" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 75%;"></div><div class="cover-empty" style="width:25%;"></div></div></td>
	<td data-value="75" class="pct medium">75%</td>
	<td data-value="4" class="abs medium">3/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="66.67" class="pct medium">66.67%</td>
	<td data-value="3" class="abs medium">2/3</td>
	<td data-value="75" class="pct medium">75%</td>
	<td data-value="4" class="abs medium">3/4</td>
	</tr>

<tr>
	<td class="file low" data-value="FavoriteRecipes.js"><a href="FavoriteRecipes.js.html">FavoriteRecipes.js</a></td>
	<td data-value="8.7" class="pic low"><div class="chart"><div class="cover-fill" style="width: 8%;"></div><div class="cover-empty" style="width:92%;"></div></div></td>
	<td data-value="8.7" class="pct low">8.7%</td>
	<td data-value="23" class="abs low">2/23</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="11" class="abs low">0/11</td>
	<td data-value="8.7" class="pct low">8.7%</td>
	<td data-value="23" class="abs low">2/23</td>
	</tr>

<tr>
	<td class="file high" data-value="GetRecipeButton.js"><a href="GetRecipeButton.js.html">GetRecipeButton.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	</tr>

<tr>
	<td class="file low" data-value="NewRecipe.js"><a href="NewRecipe.js.html">NewRecipe.js</a></td>
	<td data-value="25" class="pic low"><div class="chart"><div class="cover-fill" style="width: 25%;"></div><div class="cover-empty" style="width:75%;"></div></div></td>
	<td data-value="25" class="pct low">25%</td>
	<td data-value="4" class="abs low">1/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="3" class="abs low">0/3</td>
	<td data-value="25" class="pct low">25%</td>
	<td data-value="4" class="abs low">1/4</td>
	</tr>

<tr>
	<td class="file medium" data-value="RecipeHeader.js"><a href="RecipeHeader.js.html">RecipeHeader.js</a></td>
	<td data-value="50" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 50%;"></div><div class="cover-empty" style="width:50%;"></div></div></td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	</tr>

<tr>
	<td class="file low" data-value="RecipeIngredients.js"><a href="RecipeIngredients.js.html">RecipeIngredients.js</a></td>
	<td data-value="7.69" class="pic low"><div class="chart"><div class="cover-fill" style="width: 7%;"></div><div class="cover-empty" style="width:93%;"></div></div></td>
	<td data-value="7.69" class="pct low">7.69%</td>
	<td data-value="13" class="abs low">1/13</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="6" class="abs low">0/6</td>
	<td data-value="7.69" class="pct low">7.69%</td>
	<td data-value="13" class="abs low">1/13</td>
	</tr>

<tr>
	<td class="file high" data-value="RecipeItem.js"><a href="RecipeItem.js.html">RecipeItem.js</a></td>
	<td data-value="87.5" class="pic high"><div class="chart"><div class="cover-fill" style="width: 87%;"></div><div class="cover-empty" style="width:13%;"></div></div></td>
	<td data-value="87.5" class="pct high">87.5%</td>
	<td data-value="8" class="abs high">7/8</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="4" class="abs medium">2/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="87.5" class="pct high">87.5%</td>
	<td data-value="8" class="abs high">7/8</td>
	</tr>

<tr>
	<td class="file low" data-value="RecipeModal.js"><a href="RecipeModal.js.html">RecipeModal.js</a></td>
	<td data-value="42.86" class="pic low"><div class="chart"><div class="cover-fill" style="width: 42%;"></div><div class="cover-empty" style="width:58%;"></div></div></td>
	<td data-value="42.86" class="pct low">42.86%</td>
	<td data-value="7" class="abs low">3/7</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="4" class="abs medium">2/4</td>
	<td data-value="20" class="pct low">20%</td>
	<td data-value="5" class="abs low">1/5</td>
	<td data-value="42.86" class="pct low">42.86%</td>
	<td data-value="7" class="abs low">3/7</td>
	</tr>

<tr>
	<td class="file medium" data-value="RecipeSteps.js"><a href="RecipeSteps.js.html">RecipeSteps.js</a></td>
	<td data-value="50" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 50%;"></div><div class="cover-empty" style="width:50%;"></div></div></td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="8" class="abs medium">4/8</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="33.33" class="pct low">33.33%</td>
	<td data-value="6" class="abs low">2/6</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="8" class="abs medium">4/8</td>
	</tr>

<tr>
	<td class="file high" data-value="RecipeView.js"><a href="RecipeView.js.html">RecipeView.js</a></td>
	<td data-value="83.33" class="pic high"><div class="chart"><div class="cover-fill" style="width: 83%;"></div><div class="cover-empty" style="width:17%;"></div></div></td>
	<td data-value="83.33" class="pct high">83.33%</td>
	<td data-value="12" class="abs high">10/12</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="8" class="abs medium">4/8</td>
	<td data-value="66.67" class="pct medium">66.67%</td>
	<td data-value="3" class="abs medium">2/3</td>
	<td data-value="83.33" class="pct high">83.33%</td>
	<td data-value="12" class="abs high">10/12</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="http://istanbul-js.org/" target="_blank">istanbul</a> at Mon Jun 19 2017 17:51:46 GMT-0700 (PDT)
</div>
</div>
<script src="../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../sorter.js"></script>
</body>
</html>
