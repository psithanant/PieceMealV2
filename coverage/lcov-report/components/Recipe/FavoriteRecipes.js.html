<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for components/Recipe/FavoriteRecipes.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">components/Recipe</a> FavoriteRecipes.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">12.5% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>2/16</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/7</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">12.5% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>2/16</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90
91
92
93
94
95
96
97
98
99
100
101
102
103
104</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { Component } from 'react';
import { ListView, TouchableOpacity } from 'react-native';
import { connect } from 'react-redux';
import {
    Container,
    Content,
    Icon,
    View,
    Card,
    CardItem,
    Thumbnail,
    Text,
    Right } from 'native-base';
import RecipeModal from './RecipeModal';
import iceCream from '../../assets/food/ice-cream.jpg';
import bruscetta from '../../assets/food/Bruscetta.jpg';
import pepperBeef from '../../assets/food/pepperBeef.jpg';
&nbsp;
const images = [iceCream, bruscetta, pepperBeef];
&nbsp;
class FavoriteRecipes extends Component {
&nbsp;
  state = {
      modalVisible: false,
      name: '',
      description: '',
      notes: ''
  }
&nbsp;
<span class="fstat-no" title="function not covered" >  se</span>tModalVisible(visible) {
<span class="cstat-no" title="statement not covered" >      this.setState({ modalVisible: visible });</span>
  }
&nbsp;
<span class="fstat-no" title="function not covered" >  lo</span>adDataSource() {
      const ds = <span class="cstat-no" title="statement not covered" >new ListView.DataSource({</span>
          rowHasChanged: <span class="fstat-no" title="function not covered" >(r</span>1, r2) =&gt; <span class="cstat-no" title="statement not covered" >r1 !== r2</span>
      });
&nbsp;
<span class="cstat-no" title="statement not covered" >      this.dataSource = ds.cloneWithRows(this.props.recipes);</span>
   }
&nbsp;
<span class="fstat-no" title="function not covered" >  re</span>nderRow(rowData) {
<span class="cstat-no" title="statement not covered" >      return (</span>
          &lt;Card&gt;
              &lt;TouchableOpacity
                onPress={<span class="fstat-no" title="function not covered" >()</span> =&gt; {
<span class="cstat-no" title="statement not covered" >                this.state.name = rowData.name;</span>
<span class="cstat-no" title="statement not covered" >                this.state.description = rowData.description;</span>
<span class="cstat-no" title="statement not covered" >                this.state.notes = rowData.notes;</span>
<span class="cstat-no" title="statement not covered" >                this.state.image = images[rowData.id - 1];</span>
<span class="cstat-no" title="statement not covered" >                this.state.recipe = rowData;</span>
<span class="cstat-no" title="statement not covered" >                this.setModalVisible(true);</span>
              }}
              &gt;
                  &lt;CardItem&gt;
                      &lt;Thumbnail source={images[rowData.id - 1]} /&gt;
                      &lt;Text style={{ alignSelf: 'center', marginLeft: 25 }}&gt;{ rowData.name }&lt;/Text&gt;
                      &lt;Right&gt;
                      &lt;Icon name="arrow-forward" /&gt;
                      &lt;/Right&gt;
                  &lt;/CardItem&gt;
              &lt;/TouchableOpacity&gt;
          &lt;/Card&gt;
      );
  }
&nbsp;
<span class="fstat-no" title="function not covered" >  re</span>nder() {
<span class="cstat-no" title="statement not covered" >      this.loadDataSource();</span>
<span class="cstat-no" title="statement not covered" >          return (</span>
              &lt;Container&gt;
                  &lt;Content&gt;
                      &lt;RecipeModal
                        visible={this.state.modalVisible}
                        setModalVisible={this.setModalVisible.bind(this)}
                        name={this.state.name}
                        description={this.state.description}
                        notes={this.state.notes}
                        image={this.state.image}
                        recipe={this.state.recipe}
                        loading={this.props.isFetching}
                      /&gt;
                      &lt;View&gt;
                        &lt;ListView
                          dataSource={this.dataSource}
                          renderRow={this.renderRow.bind(this)}
                          enableEmptySections
                        /&gt;
                      &lt;/View&gt;
                  &lt;/Content&gt;
              &lt;/Container&gt;
          );
    }
}
&nbsp;
const mapStateToProps = <span class="fstat-no" title="function not covered" >(s</span>tate) =&gt; {
<span class="cstat-no" title="statement not covered" >    return {</span>
        recipes: state.clientReducer.client.recipes,
        isFetching: state.clientReducer.loading,
        user: state.loginReducer.user
    };
};
&nbsp;
export default connect(mapStateToProps)(FavoriteRecipes);
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="http://istanbul-js.org/" target="_blank">istanbul</a> at Fri Jun 16 2017 14:33:43 GMT-0700 (PDT)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
