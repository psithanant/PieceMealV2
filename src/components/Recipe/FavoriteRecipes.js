import React, { Component } from 'react';
import { ListView, TouchableOpacity } from 'react-native';
import { connect } from 'react-redux';
import {
  Button,
    Container,
    Content,
    Icon,
    View,
    Card,
    CardItem,
    Thumbnail,
    Text,
    Right } from 'native-base';
import Create from './Create';
import NewRecipe from './NewRecipe';
import RecipeModal from './RecipeModal';
import iceCream from '../../assets/food/ice-cream.jpg';
import bruscetta from '../../assets/food/Bruscetta.jpg';
import pepperBeef from '../../assets/food/pepperBeef.jpg';

const images = [iceCream, bruscetta, pepperBeef];

class FavoriteRecipes extends Component {

  state = {
      createVisible: false,
      modalVisible: false,
      name: '',
      description: '',
      notes: '',
      cookTime: '',
      prepTime: '',
      tags: []
  }

  componentDidMount() {
    return this.loadDataSource();
  }

  componentDidUpdate() {
    return this.loadDataSource();
  }

  setCreateVisible(visible) {
      this.setState({ createVisible: visible });
  }

  setModalVisible(visible) {
      this.setState({ modalVisible: visible });
  }

  loadDataSource() {
      const ds = new ListView.DataSource({
          rowHasChanged: (r1, r2) => r1 !== r2
      });

      this.dataSource = ds.cloneWithRows(this.props.recipes);
   }

  renderRow(rowData) {
      return (
          <Card style={{ marginLeft: 5, marginRight: 5 }}>
              <TouchableOpacity
                onPress={() => {
                this.state.name = rowData.name;
                this.state.description = rowData.description;
                this.state.notes = rowData.notes;
                this.state.cookTime = rowData.cook_time;
                this.state.prepTime = rowData.prep_time;
                this.state.tags = rowData.tags;
                this.state.image = images[rowData.id - 1];
                this.state.recipe = rowData;
                this.setModalVisible(true);
              }}
              >
                  <CardItem>
                      <Thumbnail source={images[rowData.id - 1]} />
                      <Text style={{ alignSelf: 'center', marginLeft: 25 }}>{ rowData.name }</Text>
                      <Right>
                      <Icon name="arrow-forward" />
                      </Right>
                  </CardItem>
              </TouchableOpacity>
          </Card>
      );
  }

  render() {
      this.loadDataSource();
          return (
              <Container>
                  <Content>
                      <RecipeModal
                        visible={this.state.modalVisible}
                        setModalVisible={this.setModalVisible.bind(this)}
                        name={this.state.name}
                        description={this.state.description}
                        cookTime={this.state.cookTime}
                        prepTime={this.state.prepTime}
                        tags={this.state.tags}
                        notes={this.state.notes}
                        image={this.state.image}
                        recipe={this.state.recipe}
                        loading={this.props.isFetching}
                      />
                      <Create
                        style={{ paddingTop: 20, marginTop: 20 }}
                        visible={this.state.createVisible}
                        setCreateVisible={this.setCreateVisible.bind(this)}
                      />
                      {/* <NewRecipe
                        style={{ paddingTop: 20, marginTop: 20 }}
                        visible={this.state.createVisible}
                        setCreateVisible={this.setCreateVisible.bind(this)}
                      /> */}
                      <View>
                        <ListView
                          dataSource={this.dataSource}
                          renderRow={this.renderRow.bind(this)}
                          enableEmptySections
                        />
                      </View>
                      <Button
                        icon
                        style={{ alignSelf: 'flex-end', marginTop: 10, marginRight: 10 }}
                        onPress={() => { this.setCreateVisible(true); }}
                      >
                        <Icon name='add' />
                      </Button>
                  </Content>
              </Container>
          );
    }
}

const mapStateToProps = (state) => {
    return {
        recipes: state.clientReducer.client.recipes,
        isFetching: state.clientReducer.loading,
        user: state.loginReducer.user
    };
};

export default connect(mapStateToProps)(FavoriteRecipes);
