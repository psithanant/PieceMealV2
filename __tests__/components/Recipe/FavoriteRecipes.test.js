import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { FavoriteRecipes } from '../../../src/components/Recipe/FavoriteRecipes';

describe('tests FavoriteRecipes', () => {
  it('renders FavoriteRecipes component', () => {
  const component = shallow(
    <FavoriteRecipes
    />);
    expect(toJson(component)).toMatchSnapshot();
  });
});
