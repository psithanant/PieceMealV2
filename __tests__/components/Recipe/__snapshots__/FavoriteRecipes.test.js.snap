// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests FavoriteRecipes renders FavoriteRecipes component 1`] = `
<Styled(Container)>
  <Styled(Content)>
    <ModalExample
      description=""
      name=""
      notes=""
      setModalVisible={[Function]}
      visible={false}
    />
    <Styled(ViewNB)>
      <ListViewMock
        enableEmptySections={true}
        renderRow={[Function]}
        renderScrollComponent={[Function]}
      />
    </Styled(ViewNB)>
  </Styled(Content)>
</Styled(Container)>
`;
